Frameworks/Foundation/NSTimeZone.mm

```diff
@@ -280,103 +127,72 @@ + (BOOL)supportsSecureCoding {
  @Status Interoperable
 */
 - (instancetype)initWithCoder:(NSCoder*)coder {
-    NSString* tempName = [coder decodeObjectForKey:@"NS.TimeZoneName"];
-    NSData* tempData = [coder decodeObjectOfClass:[NSData class] forKey:@"NS.TimeZoneData"];
-    return [self initWithName:tempName data:tempData];
+    if (self = [super init]) {
+        NSString* name = [coder decodeObjectForKey:@"NS.name"];
+        NSData* data = [coder decodeObjectOfClass:[NSData class] forKey:@"NS.data"];
+
+        return [self initWithName:name data:data];
```

Comment:
We shouldn't call both `[super init]` and `[self init ...]`; it might be more prudent to deserialize these two members and then return `[self init...]` from there.